lib/logstash/outputs/logdna.rb

```diff
@@ -122,7 +122,8 @@ def getFromHost(host,key)
   end
 
   def genLogDNAUrl(event)
-    tmp_url = "#{@base_url}?apikey=#{@api_key}"
+    ingestion_key = event.get('[@metadata][logdna][api_key]') ? event.get('[@metadata][logdna][api_key]') : @api_key
+    tmp_url = "#{@base_url}?apikey=#{ingestion_key}"
```

Comment:
Using this solution would allow end users to reference any field etc they want instead of limiting them to `[@metadata][logdna][apikey]`. They would just need to set their config value for `api_key` to `"%{@[myfield]}"`.
```suggestion
    event_api_key = event.sprintf(@api_key)
    tmp_url = "#{@base_url}?apikey=#{event_api_key }"
```

Reply:
@danf425 Looks like I was too slow, but heres an option that may be more flexible.